flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/cleanup/CheckpointResourcesCleanupRunner.java

```diff
@@ -118,7 +119,11 @@ public void start() throws Exception {
     }
 
     private void cleanupCheckpoints() throws Exception {
-        final CompletedCheckpointStore completedCheckpointStore = createCompletedCheckpointStore();
+        final CompletedCheckpointStore completedCheckpointStore =
+                // Using RestoreMode.CLAIM to be able to discard shared state, if any
+                // Note that it also means that the original shared state might be discarded as well
+                // because the initial checkpoint might be subsumed.
+                createCompletedCheckpointStore(RestoreMode.CLAIM);
```

Comment:
Could we move the `RestoreMode` into the private method `createCompletedCheckpointStore`? At least, I don't see a reason to introduce a parameter here...

Reply:
I agree, inlined the argument.